phoenix-core/src/main/java/org/apache/phoenix/hbase/index/IndexQosRpcControllerFactory.java

```diff
@@ -60,27 +61,22 @@ public PayloadCarryingRpcController newController(List<CellScannable> cellIterab
 
     private class IndexQosRpcController extends DelegatingPayloadCarryingRpcController {
 
-        private Configuration conf;
         private int priority;
 
         public IndexQosRpcController(PayloadCarryingRpcController delegate, Configuration conf) {
             super(delegate);
-            this.conf = conf;
             this.priority = PhoenixIndexRpcSchedulerFactory.getMinPriority(conf);
         }
-
         @Override
         public void setPriority(final TableName tn) {
             // if its an index table, then we override to the index priority
-            if (isIndexTable(tn)) {
+            if (!tn.isSystemTable() &&  !SchemaUtil.isSystemTable(tn.getName())) {
```

Comment:
Sorry to flip/flop on you, but I think the way you did it before was better as if we end up adding indexes to system tables, they wouldn't set the priority like we'd want them to.